File: JavaScript/옹알이(2).js
function solution(babbling) {
  let count = 0;
  let dup = ["ayaaya", "yeye", "woowoo", "mama"];

  while (babbling.length) {
    let result = babbling.pop();
    if (dup.some((v) => result.includes(v))) continue;
    result = result
      .replaceAll("aya", "1")
      .replaceAll("ye", "2")
      .replaceAll("woo", "3")
      .replaceAll("ma", "4");
    result = result.replace(/[1234]/g, "");
    if (result.length === 0) count++;
  }

  return count;
}

File: Swift/옹알이_(2).swift
import Foundation

func counting(_ babbling: String) -> Bool {
    let words = ["aya", "ye", "woo", "ma"]
    var result = ""
    var cache = ""
    
    for i in babbling.map(String.init) {
        result += i
        if words.contains(result) && result != cache {
            cache = result
            result = ""
        }
    }
    return result.isEmpty
}

func solution(_ babbling:[String]) -> Int {
    babbling.filter { counting($0) }.count
}